front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -228,193 +199,193 @@ export const usePathfinding = (
+      dispatch(
+        setWarning({
+          title: t('warningMessages.pathfindingChange'),
+          text: t('warningMessages.powerRestrictionsReset'),
+        })
+      );
+    }
+    dispatch(updatePathSteps(updatedPathSteps));
+
+    const allWaypoints = upsertPathStepsInOPs(
+      suggestedOperationalPoints,
+      compact(updatedPathSteps)
+    );
+
+    setPathProperties({
+      electrifications,
+      geometry,
+      suggestedOperationalPoints,
+      allWaypoints,
+      length: pathResult.length,
+      trackSectionRanges: pathResult.track_section_ranges,
+      incompatibleConstraints,
+    });
+  };
 
-        return {
-          ...step,
-          positionOnPath: pathResult.path_item_positions[i],
-          stopFor,
-          stopType,
-          theoreticalMargin,
-          ...(correspondingOp && {
-            name: correspondingOp.name,
-            uic: correspondingOp.uic,
-            ch: correspondingOp.ch,
-            kp: correspondingOp.kp,
-            coordinates: correspondingOp.coordinates,
-          }),
-        };
-      });
+  const launchPathfinding = async (newPathSteps?: (PathStep | null)[]) => {
+    if (newPathSteps) {
+      dispatch(replaceItinerary(newPathSteps));
+    }
+    setPathProperties(undefined);
 
-      if (!isEmptyArray(powerRestrictions)) {
-        dispatch(
-          setWarning({
-            title: t('warningMessages.pathfindingChange'),
-            text: t('warningMessages.powerRestrictionsReset'),
-          })
-        );
-      }
-      dispatch(updatePathSteps(updatedPathSteps));
+    const steps = newPathSteps || compact(pathSteps);
+    if (steps.some((step) => step === null)) {
+      dispatch(setFailure({ name: t('pathfindingError'), message: t('missingPathSteps') }));
+      return;
+    }
 
-      const allWaypoints = upsertPathStepsInOPs(
-        suggestedOperationalPoints,
-        compact(updatedPathSteps)
-      );
+    if (infra?.state !== 'CACHED') {
+      return;
+    }
 
-      setPathProperties({
-        electrifications,
-        geometry,
-        suggestedOperationalPoints,
-        allWaypoints,
-        length: pathResult.length,
-        trackSectionRanges: pathResult.track_section_ranges,
-        incompatibleConstraints,
-      });
-    };
+    pathfindingDispatch({ type: 'PATHFINDING_STARTED' });
+    const pathfindingInput = getPathfindingQuery({
+      infraId,
+      rollingStock,
+      pathSteps: steps.filter(
+        (step) => step !== null && step.coordinates !== null && !step.isInvalid
+      ),
+    });
+
+    if (!pathfindingInput) {
+      return;
+    }
 
-    const startPathFinding = async () => {
-      setPathProperties(undefined);
-      if (pathfindingState.running) {
+    try {
+      const pathfindingResult = await postPathfindingBlocks(pathfindingInput).unwrap();
+
+      if (pathfindingResult.status === 'success') {
+        await populateStoreWithPathfinding(steps as PathStep[], pathfindingResult);
```

Comment:
Why do we need a cast here? Could we replace it with:

```ts
steps.map((step) => step!)
```

which should be safer in case the type of `steps` is changed to something else?